src/gqtp_request.rs

```diff
@@ -0,0 +1,131 @@
+use std::io::Cursor;
+use std::io::prelude::*;
+use std::net::TcpStream;
+use byteorder::{BigEndian, WriteBytesExt, ReadBytesExt};
+
+const RECV_BUF_SIZE: usize = 8192;
+const GQTP_HEADER_SIZE: usize = 24;
+
+#[derive(Debug)]
+pub enum GQTPError {
+    InvalidProtocol,
+    InvalidBodySize,
+    StatusError(u16),
+}
+
+pub struct GQTPRequest<'a> {
+    addr: &'a str,
+}
+
+impl<'a> Default for GQTPRequest<'a> {
+    fn default() -> GQTPRequest<'a> {
+        GQTPRequest { addr: "127.0.0.1:10043" }
+    }
+}
+
+impl<'a> GQTPRequest<'a> {
+    /// Create a GQTP client.
+    pub fn new() -> GQTPRequest<'a> {
+        GQTPRequest::default()
+    }
+
+    /// Set host address for GQTP server.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// extern crate ruroonga_client as groonga;
+    ///
+    /// groonga::GQTPRequest::new().with_addr("127.0.0.1:20043");
+    /// ```
+    pub fn with_addr(mut self, addr: &'a str) -> GQTPRequest<'a> {
+        self.addr = addr;
+        self
+    }
+
+    /// Send request and Receive response.
+    pub fn call<C>(&self, command: C) -> Result<String, GQTPError>
```

Comment:
Off topic.
I want to split this method into write and read.
But this implementation is OK for now.